cmd/zc_filter.go

```diff
@@ -59,6 +59,39 @@ func (f *excludeBlobTypeFilter) DoesPass(object StoredObject) bool {
 	return false
 }
 
+// excludeContainerFilter filters out container names that must be excluded
+type excludeContainerFilter struct {
+	containerNamesList []string
+}
+
+func (s *excludeContainerFilter) DoesSupportThisOS() (msg string, supported bool) {
+	return "", true
+}
+
+func (s *excludeContainerFilter) AppliesOnlyToFiles() bool {
+	return false // excludeContainerFilter is related to container names, not related to files
+}
+
+func (s *excludeContainerFilter) DoesPass(storedObject StoredObject) bool {
+	if len(s.containerNamesList) == 0 {
+		return true
+	}
+	for _, cName := range s.containerNamesList {
+		// container name matches the excluded container name and return false to exclude
+		if strings.Compare(cName, storedObject.ContainerName) == 0 {
+			return false
+		}
+	}
+	return true
+}
+
+func buildExcludeContainerFilter(containerNames []string) []ObjectFilter {
+	excludeContainerSet := make([]string, 0)
+	excludeContainerSet = append(excludeContainerSet, containerNames...)
+
+	return append(make([]ObjectFilter, 0), &excludeContainerFilter{containerNamesList: excludeContainerSet})
```

Comment:
For the above comment, here instead of a slice we can just make a map of strings.

Reply:
Updated :)